contracts/beacon/Voucher.sol

```diff
@@ -196,4 +238,50 @@ contract Voucher is OwnableUpgradeable, IVoucher {
             $.slot := VOUCHER_STORAGE_LOCATION
         }
     }
+
+    /**
+     * @dev Debit voucher and transfer non-sponsored amount from requester's account.
+     *
+     * @param appOrder The app order.
+     * @param datasetOrder The dataset order.
+     * @param workerpoolOrder The workerpool order.
+     * @param requestOrder The request order.
+     *
+     * @return sponsoredAmount The amount sponsored by the voucher.
+     */
+    function _debitVoucherAndTransfer(
```

Comment:
Minor:
```suggestion
    function _debitVoucherAndTransferNonSponsoredAmount(
```
You version is better since short
This suggestion is better since bit more meaningful
I don't know which one is the best :smile: (maybe yours or other!)

Reply:
Since the function is `private` it will be more understandable to use your name I think